designs/isolate-cluster-upgrades.md

```diff
@@ -0,0 +1,221 @@
+# Isolate workload clusters from management cluster upgrades
+
+## Introduction
```

Comment:
Would machine rollout being triggered when you upgrade a cluster-api controller? If so, adding "generation" would not prevent this kind of machine rollout from happening